### PR TITLE
ARCH-1617 - Implementing the initial action

### PR DESCRIPTION
**Pull request by danielle-casella-adams**
# Summary of PR changes

-  Implementing the action

## PR Requirements
- [x] For major, minor, or breaking changes, at least one of the commit messages contains the appropriate `+semver:` keywords.
  - See the *Incrementing the Version* section of the repository's README.md for more details.
- [x] The examples in the repository's `README.md` have been updated with the new version.  
  - See the *Incrementing the Version* section of the repository's README.md for more details on how the version will be incremented.
- [x] The action code does not contain sensitive information.

### CODE REVIEW

**Review by hpractv — APPROVED** [review]
👍🏽

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -18,20 +21,50 @@ jobs:
...
           node-version: 16.x
 
-      - name: Build
+      - name: Recompile and Format action
         run: npm run build
 
-      - name: Format
-        run: npm run format
+      # These last steps only need to be done once
+      # per commit.  Run them on the ubuntu job.
```
Comment (JosephDSchwartz) [inline .github/workflows/build.yml:40]: I may have missed something, but the steps below don't look like they reference anything from the steps above. So out of curiosity, could they be put into a different job that relies on the success of this matrix job? Or is that more effort/confusing/etc.?

Comment (danielle-casella-adams) [reply .github/workflows/build.yml:40]: The commit step would depend on the recompile and format step (which happens on all three matrix jobs) if there are outstanding changes from there.

Comment (JosephDSchwartz) [reply .github/workflows/build.yml:40]: Oh right, good point. I guess we could pull out ubuntu from the matrix and make it a separate job, but that would look a little weird.

**Review by JosephDSchwartz — APPROVED** [review]

**Pull request merged**